src/m_misc.c

```diff
@@ -48,14 +47,173 @@
 #include "w_wad.h"
 #include "z_zone.h"
 
+#ifdef _WIN32
+static wchar_t* ConvertToUtf8(const char *str)
+{
+    wchar_t *wstr = NULL;
+    int wlen = 0;
+
+    wlen = MultiByteToWideChar(CP_UTF8, 0, str, -1, NULL, 0);
+
+    if (!wlen)
+    {
+        printf("Warning: Failed to convert path to UTF8\n");
+        return NULL;
+    }
+
+    wstr = malloc(sizeof(wchar_t) * wlen);
+
+    if (!wstr)
+    {
+        I_Error("ConvertToUtf8: Failed to allocate new string");
+        return NULL;
+    }
+
+    if (MultiByteToWideChar(CP_UTF8, 0, str, -1, wstr, wlen) == 0)
+    {
+        printf("Warning: Failed to convert path to UTF8\n");
+        free(wstr);
+        return NULL;
+    }
+
+    return wstr;
+}
+#endif
+
+FILE* M_fopen(const char *filename, const char *mode)
+{
+#ifdef _WIN32
+    FILE *file;
+    wchar_t *wname = NULL;
+    wchar_t *wmode = NULL;
+
+    wname = ConvertToUtf8(filename);
+
+    if (!wname)
+    {
+        return NULL;
+    }
+
+    wmode = ConvertToUtf8(mode);
+
+    if (!wmode)
+    {
+        free(wname);
+        return NULL;
+    }
+
+    file = _wfopen(wname, wmode);
+
+    free(wname);
+    free(wmode);
+
+    return file;
+#else
+    return fopen(filename, mode);
+#endif
+}
+
+int M_remove(const char *path)
+{
+#ifdef _WIN32
+    wchar_t *wpath = NULL;
+    int ret;
+
+    wpath = ConvertToUtf8(path);
+
+    if (!wpath)
+    {
+        return 0;
+    }
+
+    ret = _wremove(wpath);
+
+    free(wpath);
+
+    return ret;
+#else
+    return remove(path);
+#endif
+}
+
+int M_rename(const char *oldname, const char *newname)
+{
+#ifdef _WIN32
+    wchar_t *wold = NULL;
+    wchar_t *wnew = NULL;
+    int ret;
+
+    wold = ConvertToUtf8(oldname);
+
+    if (!wold)
+    {
+        return 0;
+    }
+
+    wnew = ConvertToUtf8(newname);
+
+    if (!wnew)
+    {
+        free(wold);
+        return 0;
+    }
+
+    ret = _wrename(wold, wnew);
+
+    free(wold);
+    free(wnew);
+
+    return ret;
+#else
+    return rename(oldname, newname);
+#endif
+}
+
+int M_stat(const char *path, struct stat *buf)
+{
+#ifdef _WIN32
+    wchar_t *wpath = NULL;
+    struct _stat wbuf;
+    int ret;
+
+    wpath = ConvertToUtf8(path);
+
+    if (!wpath)
+    {
+        return 0;
```

Comment:
This is wrong, according to documentation `stat` will return 0 on success and -1 on error. `errno` should also be set if the UTF conversion function doesn't do it.